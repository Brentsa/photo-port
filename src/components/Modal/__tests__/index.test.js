import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import Modal from '..';

afterEach(cleanup);

const currentPhoto = {
    name: 'Park bench',
    category: 'landscape',
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie',
    index: 1
}

const mockSetModalIsOpen = jest.fn();

describe('Modal component renders', ()=>{
    it('Renders modal', () => {
        render(<Modal currentPhoto={currentPhoto} setIsModalOpen={mockSetModalIsOpen}/>);
    });

    it('Matches snapshot', () => {
        const {asFragment} = render(<Modal currentPhoto={currentPhoto} setIsModalOpen={mockSetModalIsOpen}/>);
        expect(asFragment).toMatchSnapshot();
    });
});

describe('Click Event', () => {
    it('calls onClose handler', () => {
        const {getByText} = render(<Modal currentPhoto={currentPhoto} setIsModalOpen={mockSetModalIsOpen}/>);
        fireEvent.click(getByText('Close this modal'));
        expect(mockSetModalIsOpen).toHaveBeenCalled();
    })
})