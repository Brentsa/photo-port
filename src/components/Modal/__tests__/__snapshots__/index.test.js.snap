// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component renders Matches snapshot 1`] = `[Function]`;
